lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1267,6 +1267,32 @@ LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II, SPIRVBasicBlock *BB) {
       transValue(II->getOperand(2), BB),
       getMemoryAccess(cast<MemIntrinsic>(II)),
       BB);
+  case Intrinsic::lifetime_start : {
+    int64_t Size = dyn_cast<ConstantInt>(II->getOperand(0))->getSExtValue();
+    if (Size == -1)
+      Size = 0;
+    if (isa<BitCastInst>(II->getOperand(1))) {

```

Comment:
Need to check the size of the real object matches the size argument of lifetime.start, otherwise cannot drop the bitcast.